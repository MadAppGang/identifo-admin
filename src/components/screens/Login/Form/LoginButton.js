import React from 'react';
import PropTypes from 'prop-types';
import loadingIcon from './loading.svg';

const LoginButton = ({ loading, onClick }) => (
  <button
    type="button"
    className="iap-login-form__submit-btn"
    disabled={loading}
    onClick={onClick}
  >
    {loading && (
      <img
        src={loadingIcon}
        alt="spinner"
        className="iap-login-form__submit-indicator"
      />
    )}
    Sign In
  </button>
);

LoginButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

LoginButton.defaultProps = {
  loading: false,
};

export default LoginButton;
