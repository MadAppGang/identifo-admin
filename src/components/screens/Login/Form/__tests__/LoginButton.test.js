import LoginButton from '../LoginButton';

describe('<LoginButton />', () => {
  test('renders as expected', () => {
    expect(shallow(<LoginButton onClick={jest.fn()} />)).toMatchSnapshot();
  });

  test('renders a loading indicator when loading is passed in as true', () => {
    expect(shallow(<LoginButton onClick={jest.fn()} loading />)).toMatchSnapshot();
  });

  test('calls onClick prop on button click', () => {
    const onClick = jest.fn();

    shallow(<LoginButton onClick={onClick} />)
      .find('button')
      .simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
