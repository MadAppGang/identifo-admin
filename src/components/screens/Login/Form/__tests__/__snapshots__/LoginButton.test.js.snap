// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> renders as expected 1`] = `
<Button
  disabled={false}
  icon={null}
  onClick={[MockFunction]}
  stretch={true}
  transparent={false}
  type="submit"
>
  Sign In
</Button>
`;
